<a href="https://colab.research.google.com/github/J878-commits/-Task-1-Text-Summarization-with-Transformers-Gradio-/blob/main/CodTech_Internship_%E2%80%93_Task_3_Convolutional_Neural_Network_for_CIFAR_10.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

🧠 Project Overview: CNN for Image Classification

🎯 Goal
Build and evaluate a CNN that can classify images into categories (e.g., animals, objects, digits, etc.) using a standard dataset like CIFAR-10, MNIST, or a custom dataset if specified.

🛠️ Step-by-Step Workflow
1. Setup & Imports
Use TensorFlow (recommended for smoother deployment and integration with your existing work).

In [1]:
import tensorflow as tf
from tensorflow.keras import layers, models
import matplotlib.pyplot as plt


2. Load Dataset
Use CIFAR-10 for a balanced, colorful image dataset with 10 classes.

In [2]:
(x_train, y_train), (x_test, y_test) = tf.keras.datasets.cifar10.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0  # Normalize


170498071/170498071 ━━━━━━━━━━━━━━━━━━━━ 2s 0us/step


3. Build the CNN Model

In [3]:
model = models.Sequential([
    layers.Conv2D(32, (3, 3), activation='relu', input_shape=(32, 32, 3)),
    layers.MaxPooling2D((2, 2)),
    layers.Conv2D(64, (3, 3), activation='relu'),
    layers.MaxPooling2D((2, 2)),
    layers.Conv2D(64, (3, 3), activation='relu'),
    layers.Flatten(),
    layers.Dense(64, activation='relu'),
    layers.Dense(10, activation='softmax')
])


/usr/local/lib/python3.11/dist-packages/keras/src/layers/convolutional/base_conv.py:113: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


4. Compile & Train

In [5]:
model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

history = model.fit(x_train, y_train, epochs=5,
                    validation_data=(x_test, y_test))


Epoch 1/5
1563/1563 ━━━━━━━━━━━━━━━━━━━━ 71s 44ms/step - accuracy: 0.6721 - loss: 0.9227 - val_accuracy: 0.6564 - val_loss: 0.9781
Epoch 2/5
1563/1563 ━━━━━━━━━━━━━━━━━━━━ 81s 43ms/step - accuracy: 0.6995 - loss: 0.8517 - val_accuracy: 0.6827 - val_loss: 0.9169
Epoch 3/5
1563/1563 ━━━━━━━━━━━━━━━━━━━━ 83s 44ms/step - accuracy: 0.7169 - loss: 0.8061 - val_accuracy: 0.6938 - val_loss: 0.8975
Epoch 4/5
1563/1563 ━━━━━━━━━━━━━━━━━━━━ 82s 44ms/step - accuracy: 0.7293 - loss: 0.7614 - val_accuracy: 0.6778 - val_loss: 0.9520
Epoch 5/5
1563/1563 ━━━━━━━━━━━━━━━━━━━━ 82s 44ms/step - accuracy: 0.7504 - loss: 0.7130 - val_accuracy: 0.6966 - val_loss: 0.8849


5. Evaluate Performance

In [6]:
test_loss, test_acc = model.evaluate(x_test, y_test)
print(f"Test Accuracy: {test_acc:.2f}")


313/313 ━━━━━━━━━━━━━━━━━━━━ 4s 12ms/step - accuracy: 0.6993 - loss: 0.8800
Test Accuracy: 0.70


Based on the notebook, the task is to build and evaluate a CNN for image classification using the CIFAR-10 dataset. The model was trained for 5 epochs and achieved a test accuracy of approximately 70%.

✅ Task 3 Summary: CNN for Image Classification

Objective: Build and evaluate a Convolutional Neural Network (CNN) for image classification using the CIFAR-10 dataset, as part of the CodTech internship deliverables.

Approach:

Framework: TensorFlow

Dataset: CIFAR-10 (60,000 32×32 color images in 10 classes)

Model: Custom CNN architecture with Conv2D, MaxPooling, Flatten, Dense layers

Training: 5 epochs

Evaluation: model.evaluate(x_test, y_test)

Results:

Test Accuracy: ~70%

Test Loss: ~0.88

Model successfully classifies images across 10 categories with reasonable accuracy, meeting the internship’s functional and evaluation criteria.

Status: ✅ Task 3 Completed — Ready for certificate issuance upon internship end date.